File: backend/app/schemas/moment.py
"""朋友圈请求/响应模型"""
from datetime import datetime
from pydantic import BaseModel, Field


class MomentResponse(BaseModel):
    id: int
    character_id: int = 0
    character_name: str = ""
    avatar_url: str | None = None
    user_id: int = 0
    sender_type: str = "ai"  # ai / user
    content: str
    image_url: str | None = None
    image_desc: str | None = None
    likes_count: int
    is_active: bool
    created_at: datetime
    author_tz_offset: int = 8  # 作者所在时区（UTC 偏移小时，朋友圈时间按作者地区显示）
    liked_by_me: bool = False
    likers: list[str] = []
    comments: list["CommentResponse"] = []

    model_config = {"from_attributes": True}


class MomentListResponse(BaseModel):
    moments: list[MomentResponse]
    total: int


class LikeResponse(BaseModel):
    moment_id: int
    likes_count: int
    liked: bool


class CommentResponse(BaseModel):
    id: int
    moment_id: int
    parent_id: int | None = None
    sender_type: str
    sender_id: int
    sender_name: str
    content: str
    created_at: datetime
    replies: list["CommentResponse"] = []

    model_config = {"from_attributes": True}


class CommentListResponse(BaseModel):
    comments: list[CommentResponse]
    total: int


class CreateCommentRequest(BaseModel):
    content: str = Field(..., min_length=1, max_length=500)
    parent_id: int | None = None


class CreateUserMomentRequest(BaseModel):
    content: str = Field(..., min_length=1, max_length=2000)

File: flutter_app/lib/models/moment.dart
import 'moment_comment.dart';

class Moment {
  final int id;
  final int characterId;
  final String characterName;
  final int userId;
  final String senderType;
  final String? avatarUrl;
  final String content;
  final String? imageUrl;
  final String? imageDesc;
  final int likesCount;
  final List<String> likers;
  final bool isActive;
  final String createdAt;
  final int authorTzOffset;
  final bool likedByMe;
  final List<MomentComment> comments;

  Moment({
    required this.id, required this.characterId, required this.characterName,
    required this.userId, required this.senderType, this.avatarUrl, required this.content,
    this.imageUrl, this.imageDesc,
    required this.likesCount, this.likers = const [], required this.isActive, required this.createdAt,
    this.authorTzOffset = 8,
    required this.likedByMe,
    this.comments = const [],
  });

  factory Moment.fromJson(Map<String, dynamic> json) {
    return Moment(
      id: json['id'] as int,
      characterId: json['character_id'] as int? ?? 0,
      characterName: json['character_name'] as String? ?? "",
      userId: json['user_id'] as int? ?? 0,
      senderType: json['sender_type'] as String? ?? "ai",
      avatarUrl: json['avatar_url'] as String?,
      content: json['content'] as String,
      imageUrl: json['image_url'] as String?,
      imageDesc: json['image_desc'] as String?,
      likesCount: json['likes_count'] as int? ?? 0,
      likers: (json['likers'] as List?)?.map((e) => e.toString()).toList() ?? [],
      isActive: json['is_active'] as bool? ?? true,
      createdAt: json['created_at'] as String? ?? "",
      authorTzOffset: json['author_tz_offset'] as int? ?? 8,
      likedByMe: json['liked_by_me'] as bool? ?? false,
      comments: (json['comments'] as List<dynamic>?)
              ?.map((c) => MomentComment.fromJson(c as Map<String, dynamic>))
              .toList() ?? [],
    );
  }

  Moment copyWith({int? likesCount, List<String>? likers, bool? likedByMe, List<MomentComment>? comments}) {
    return Moment(
      id: id, characterId: characterId, characterName: characterName,
      userId: userId, senderType: senderType, avatarUrl: avatarUrl, content: content,
      imageUrl: imageUrl, imageDesc: imageDesc,
      likesCount: likesCount ?? this.likesCount, likers: likers ?? this.likers, isActive: isActive,
      createdAt: createdAt, authorTzOffset: authorTzOffset, likedByMe: likedByMe ?? this.likedByMe, comments: comments ?? this.comments,
    );
  }
}